R/utils.R

```diff
@@ -1154,3 +1154,11 @@ longest_element <- function(x) {
   else
     ""
 }
+
+# Apply rlang::as_quosure across a list or vector, but explicitly pass env
+# and expr to as_quosure, due to a bug in as_quosures when dealing with 
+# lazy objects
+as_quosures_explicit <- function(vec) {
+  lapply(vec, 
+         FUN = function(x) rlang::as_quosure(x$expr, x$env))
+}
```

Comment:
I think I'd want to make it more explicit that this function is for coercing dots to quosures


```r
dots_as_quosures <- function(x) {
  if (!inherits(x, "lazy_dots")) {
    stop("Expected lazy dots")
  }
  lapply(vec, function(x) rlang::as_quosure(x$expr, x$env))
}
```

Reply:
Addressed this with 4d363d665e30e9e27fb0b2b2dc09d1eb89ebb1e7. 

Big thing: in the rlang issue, lionel recommended using compat-lazy ([see source code here](https://github.com/r-lib/rlang/blob/master/R/compat-lazyeval.R)). It's not in rlang, but an additional ~50 lines that we'd put in utils. It's more generalizable, but more code. 

Should I put compat-lazy into utils, or is dots_as_quosures sufficient?